### PR TITLE
Add URI Handling and invite code

### PR DESCRIPTION
**Pull request by emma-miler**

### CODE REVIEW

**Comment by emma-miler** [comment]
Still needs the following things before merge:

- [ ] Make autofocus on invite
- [ ] Generating invites when joined via invite

**Comment by emma-miler** [comment]
https://github.com/R2Northstar/NorthstarMasterServer/pull/45

**Comment by abarichello** [comment]
I would like @pg9182 or someone with Windows API experience to review this before merging

File: LauncherInjector/main.cpp [hunk 1]
```diff
@@ -336,6 +396,40 @@ int main(int argc, char* argv[])
 	}
 
 	{
+
+		wchar_t buffer[_MAX_PATH];
+		GetModuleFileNameW(NULL, buffer, _MAX_PATH); // Get full executable path
+		std::wstring w = std::wstring(buffer);
+		std::string exepath = std::string(w.begin(), w.end());			   // Convert from wstring to string
+		std::string path = exepath.substr(0, exepath.find_last_of("/\\")); // Substr to just the folder name
```
Comment (emma-miler) [inline LauncherInjector/main.cpp:404]: This should check for the last `/` **before the first `.exe`**

Do not merge before this is fixed

**Review by pg9182 — COMMENTED** [review]
Not finished reviewing yet, but some quick things I noticed.

File: LauncherInjector/main.cpp (hunk 1 above)
Comment (emma-miler) [reply LauncherInjector/main.cpp:404]: Upon testing, this seems to be fine

File: NorthstarDedicatedTest/dllmain.cpp [hunk 2]
```diff
@@ -42,12 +42,16 @@
...
 #include "plugins.h"
+#include "urihandler.h"
 
 #include "rapidjson/document.h"
 #include "rapidjson/stringbuffer.h"
 #include "rapidjson/writer.h"
 #include "rapidjson/error/en.h"
 
+#include <shellapi.h>
+#include <WinUser.h>
```
[the remark this answers is not in the document]
Comment (emma-miler) [reply NorthstarDedicatedTest/dllmain.cpp:53]: I'll remove this, but i'm not sure where you found that. Could you link what docs say that?

[the remark this answers is not in the document]
Comment (pg9182) [reply NorthstarDedicatedTest/dllmain.cpp]: https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-messageboxa

File: NorthstarDedicatedTest/urihandler.cpp [hunk 3]
```diff
@@ -0,0 +1,417 @@
...
+
+SQRESULT SQ_GetInviteError(void* sqvm)
+{
+	ClientSq_pushstring(sqvm, lastError.c_str(), -1);
+	return SQRESULT_NOTNULL;
+}
+
+void StartUriHandler()
+{
+	spdlog::info("Creating an instance of a named pipe...\n");
```
Comment (pg9182) [inline NorthstarDedicatedTest/urihandler.cpp:305]: There's quite a few calls to `spdlog::info` with an unnecessary newline in this PR. Perhaps it was originally a `printf` and the newline was missed when replacing it with spdlog?

**Review by pg9182 — COMMENTED** [review]
A few more things. I haven't checked the squirrel stuff since I'm not familiar enough with it, but I've looked at the code in general and the curl + win32 stuff. I'll test it on wine tomorrow.

**Comment by GeckoEidechse** [comment]
Merge conflicts :trollface:

**Pull request closed without merging**